File: 0011-container-with-most-water/0011-container-with-most-water.js
var maxArea = function(height) {
    let i = 0;
    let j = height.length-1;
    let maxarea =0;
    
    while(i<j){
        let mh = Math.min(height[i] , height[j]);
        let d = j-i;
        let area = mh * d;
        maxarea = Math.max(maxarea, area);
        if(height[i] < height[j]){
            i++;
        }
        else{
            j--;
        }
        
    }
    return maxarea;  
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length-1;
        int maxarea=0;
        while(i<j){
            int mh = Math.min(height[i], height[j]);
            int d = j-i;
            int area = mh * d;
            maxarea = Math.max(maxarea, area);
            if(height[i]< height[j]){
                i++;
            }
            else{
                j--;
            }

        }
        return maxarea;
        
    }
}